pubtools/pulplib/_impl/fake/client.py

```diff
@@ -79,6 +82,35 @@ def get_repository(self, repository_id):
 
         return f_return(data[0])
 
+    def _do_upload_file(self, upload_id, file_obj, name):
+        # pylint: disable=unused-argument
+        out = f_return((hashlib.sha256().hexdigest(), random.randint(10, 1000)))
+
+        if "close" not in dir(file_obj):
+            file_obj = open(file_obj, "rb")
+            out.add_done_callback(lambda _: file_obj.close())
+
+        return out
+
+    def _request_upload(self):
+        upload_request = {
+            "_href": "/pulp/api/v2/content/uploads/%s/" % self._request_id(),
+            "upload_id": "%s" % self._request_id,
+        }
+
+        return f_return(upload_request)
+
+    def _do_import(self, repo_id, upload_id, unit_type_id, unit_key):
+        # pylint: disable=unused-argument
+        task = Task(id=self._next_task_id(), completed=True, succeeded=True)
+
+        self._upload_history.append(Upload(repo_id, [task]))
+
+        return f_return([task])
+
+    def _delete_upload_request(self):
+        return f_return()  # pragma: no cover
```

Comment:
Why no cover?  If this is never called then you don't even need to implement it.

I guess you really don't need to implement it, as deleting upload requests is an implementation detail not exposed in the API of the real client in any way, so there's no reason for the fake client to have any code for it.

Reply:
hm, I felt I needed it since it's called in FileRepository.upload_file, though it's not consistent.
 
By consistent I mean, since we always return the import task, so the call of upload_file().result() can't promise _delete_upload_request is called, at least in the real cases. I'm not sure if it will be called in the fake client, I'll do some test

Reply:
Turns about it's never called. I removed it. But I don't know why it's never called.